allauth/utils.py

```diff
@@ -70,12 +78,12 @@ def generate_unique_username(txts, regex=None):
                 pfx = str(i + 1)
             else:
                 pfx = ''
-            ret = username[0:max_length - len(pfx)] + pfx
-            query = {USER_MODEL_USERNAME_FIELD + '__iexact': ret}
-            User.objects.get(**query)
+            ret = base_username[0:max_length - len(pfx)] + pfx
+            username = adapter.clean_username(ret)
+            if username:
```

Comment:
This check should not be necessary... if clean does not raise a `ValidationError` it must return a valid username...